Dockerfile

```diff
@@ -61,7 +63,7 @@ ARG ADD_PIP_PACKAGES=""
 ENV TZ=${TIMEZONE} \
 	DEBIAN_FRONTEND="noninteractive" \
 	DEB_BUILD_DEPS="tzdata build-essential python3-setuptools python3-pip apt-utils git" \
-	DEB_PACKAGES="locales libgdal20 python3-gdal libsqlite3-mod-spatialite curl ${ADD_DEB_PACKAGES}" \
+	DEB_PACKAGES="locales libgdal-dev python3-gdal libsqlite3-mod-spatialite curl ${ADD_DEB_PACKAGES}" \
```

Comment:
IMO the -dev package is not required (maybe that also adds to Image size?). As it looks this should be the GDAL package: https://packages.debian.org/bullseye/libgdal26 so `libgdal26` which is strangely GDAL 3.0.4. Have you tried already?